RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift

```diff
@@ -91,18 +99,48 @@ extension HomeScreenViewController {
 
 }
 
-extension HomeScreenViewController: UICollectionViewDataSource {
+// MARK: Recommendation Collection View
+extension HomeScreenViewController {
 
-    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return viewModel.numberOfItemsInSection
+    func setupRecommendationsCollectionView() {
+        recommendationsCollectionView.delegate = self
+        recommendationsCollectionView.dataSource = self
+        recommendationsCollectionView.backgroundColor = .clear
+        recommendationsCollectionView.showsHorizontalScrollIndicator = false
+        recommendationsCollectionView.collectionViewLayout = UICollectionViewFlowLayout()
+        if let layout = recommendationsCollectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            layout.scrollDirection = .horizontal
+            layout.minimumInteritemSpacing = 0
+            layout.minimumLineSpacing = 16
+            layout.sectionInset = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 16)
+        }
     }
+}
+
+extension HomeScreenViewController: UICollectionViewDataSource {
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        numberOfItemsInSection section: Int) -> Int {
+            if collectionView == self.categoriesCollectionView {
+
+                return viewModel.categoriesNumberOfItemsInSection
+            }
+            return viewModel.recommendationsNumberOfItemsInSection
+        }
 
     func collectionView(_ collectionView: UICollectionView,
                         cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell: CategoriesCell = categoriesCollectionView.dequeue(for: indexPath)
-        let category = viewModel.categories[indexPath.row]
-        cell.setup(with: category)
-        return cell
+        if collectionView == self.categoriesCollectionView {
+            let cell: CategoriesCell = categoriesCollectionView.dequeue(for: indexPath)
+            let category = viewModel.categories[indexPath.row]
+            cell.setup(with: category)
+            return cell
+        } else {
```

Comment:
Same as my previous comment, use an `if` statement for each collection view. Also, `self` is not needed.